controllers/selfnoderemediation_controller.go

```diff
@@ -44,6 +44,7 @@ import (
 const (
 	SNRFinalizer          = "self-node-remediation.medik8s.io/snr-finalizer"
 	fencingCompletedPhase = "Fencing-Completed"
+	rebootCompletedPhase  = "Reboot-Completed"
```

Comment:
Nit: I'd consider turning into a new type and replace both `is` methods with a single generic one.
```
type remediationPhase string

var (
	fencingCompletedPhase remediationPhase = "Fencing-Completed"
	rebootCompletedPhase  remediationPhase = "Reboot-Completed"
)

func (r *SelfNodeRemediationReconciler) isAtRemediationPhase(snr *v1alpha1.SelfNodeRemediation, phase remediationPhase) bool {
	return snr.Status.Phase != nil && *snr.Status.Phase == string(phase)
}
```